bridge/pier/syncer.go

```diff
@@ -234,22 +270,22 @@ func (syncer *Syncer) processHeader(newHeader *types.Header) {
 
 		syncer.Logger.Debug("Got last block from bridge storage", "lastBlock", string(lastBlockBytes))
 		if result, err := strconv.ParseUint(string(lastBlockBytes), 10, 64); err == nil {
-			if result >= newHeader.Number.Uint64() {
-				return
+			if result > fromBlock.Uint64() {
+				fromBlock = big.NewInt(0).SetUint64(result)
 			}
 
 			fromBlock = big.NewInt(0).SetUint64(result + 1)
 		}
 	}
 
 	// to block
-	toBlock := latestNumber
+	toBlock := end
 
 	// debug log
-	syncer.Logger.Debug("Processing header", "fromBlock", fromBlock, "toBlock", toBlock)
+	syncer.Logger.Info("Processing header", "fromBlock", fromBlock, "toBlock", toBlock)
 
 	// set diff
-	if toBlock.Uint64() < fromBlock.Uint64() {
+	if toBlock.Cmp(fromBlock) == -1 {
```

Comment:
This should never happen. If it does something is wrong in the logic above.

Reply:
Right. Will remove.